### DISCUSSION TITLE
Logging and using pretext-cli as a library

### DISCUSSION THREAD

**Discussion by StevenClontz** [PreTeXtBook/pretext-cli#470] [Ideas]
Mostly for @oscarlevin - right now almost no one uses pretext-cli as a library, but the current refactor will make incorporating pretext as a dependency for other libraries much less painful.

One pain point that exists now: last time I used the Project class outside of calling the Click CLI, none of the logging.whatevers were printed to stdout. What's the right way to fix this? Right now I'm [cheating](https://github.com/PreTeXtBook/pretext-cli/blob/6cef2d6cf5b32981cf4decec7db2c93496e6f3b4/pretext/project_refactor.py#L490):

```python
def warn(self, log_warning: t.Callable = print) -> None:
    log_warning("foo")
```

But surely there's a more appropriate way to use the logging library to avoid print statements anywhere, but still output desired feedback to users even if they aren't using the CLI endpoint.

**Comment by oscarlevin** [comment]
There is definitely a way to do this with the logging package (their documentation discusses best practices for this).  I suspect I just have something set up wrong.  I should have some time to experiment with this early next week.

Also, this will close #418 when it gets closed.

**Reply by oscarlevin** [reply]
Okay, the logging should be fixed, as of #472.

I think logging can still be improved though.  It seems like the recommended best practice is to use `__name__` as the logging name, but I'm not sure what all that will change.  I think that once the refactor of project.py is complete, we can start looking at documenting how to use the CLI as a module in general, including how to modify the logs that it creates.
